src/main/java/com/sequenceiq/cloudbreak/service/stack/handler/StackDeleteCompleteHandler.java

```diff
@@ -55,6 +59,7 @@ public void accept(Event<StackDeleteComplete> stackDeleteComplete) {
         cloudbreakEventService.fireCloudbreakEvent(data.getStackId(), BillingStatus.BILLING_STOPPED.name(), BILLING_STOPPED_MSG);
         updateStackFields(stack);
         retryingStackUpdater.updateStack(stack);
+        cloudbreakEventRepository.delete(cloudbreakEventRepository.findCloudbreakEventsForStack(stack.getId()));
```

Comment:
If we delete events here, the usage generation will fail every time, because the usage generation use the cloudbreak events to calculate usages. We should delete events after the usage generation of the appropriate stack if it is in DELETE_COMPLETED status, instead.